MQ2FeedMe.cpp

```diff
@@ -79,38 +253,50 @@ void ReadList(std::list<std::string>* MyList, PCHAR fSec)
 {
 	char Buffer[MAX_STRING*10];
 	MyList->clear();
-	if(GetPrivateProfileString(fSec,NULL,"",Buffer,MAX_STRING*10,INIFileName)) {
+
+	if(GetPrivateProfileString(fSec, NULL, "", Buffer, MAX_STRING * 10 , INIFileName))
+	{
 		char  szTemp[MAX_STRING];
-		PCHAR pBuffer=Buffer;
-		while (pBuffer[0]!=0) {
-			GetPrivateProfileString(fSec,pBuffer,"",szTemp,MAX_STRING,INIFileName);
-			if(szTemp[0]!=0) MyList->push_back(std::string(szTemp));
-			pBuffer+=strlen(pBuffer)+1;
+		char* pBuffer = Buffer;
+
+		while (pBuffer[0] !=0 )
+
+		{
+			GetPrivateProfileString(fSec, pBuffer, "", szTemp, MAX_STRING, INIFileName);
+			if (szTemp[0] != 0)
+			{
+				MyList->push_back(std::string(szTemp));
+			}
+
+			pBuffer += strlen(pBuffer) + 1;
 		}
 	}
 }
 
-bool GoodToFeed()
+bool GoodToConsume()
 {
-	auto pChar = GetCharInfo();
 	auto pChar2 = GetPcProfile();
+	int iZoneID = pLocalPlayer->GetZoneID() & 0x7FFF;
 
-	if(GetGameState() == GAMESTATE_INGAME &&                  // currently ingame
-	   pChar &&                                               // have Charinfo
-	   pChar->pSpawn &&                                       // have a Spawn
-	   pChar2 &&                                              // have PcProfile
-	   !ItemOnCursor() &&                                     // nothing on cursor
-	   (!IsCasting() || pChar2->Class == Bard) &&             // not casting unless bard
-	   (!AbilityInUse() || pChar2->Class == Bard) &&          // not using abilities unless bard
-	   !WindowOpen("SpellBookWnd") &&                         // not looking at the book
-	   !WindowOpen("MerchantWnd") &&                          // not interacting with vendor
-	   !WindowOpen("TradeWnd") &&                             // not trading with someone
-	   !WindowOpen("BigBankWnd") && !WindowOpen("BankWnd") && // not banking
-	   !WindowOpen("LootWnd") &&                              // not looting
-	   pChar->pSpawn->StandState != STANDSTATE_FEIGN &&       // not Feigned
-	   !IAmCamping) {                                         // not camping
+	if(GetGameState() == GAMESTATE_INGAME						// currently ingame
+		&& pLocalPC												// have Charinfo
+		&& pLocalPC->pSpawn										// have a Spawn
+		&& pChar2												// have PcProfile
+		&& !ItemOnCursor()										// nothing on cursor
+		&& (!IsCasting() || pChar2->Class == Bard)				// not casting unless bard
+		&& (!AbilityInUse() || pChar2->Class == Bard)			// not using abilities unless bard
+		&& !WindowOpen("SpellBookWnd")							// not looking at the book
+		&& !WindowOpen("MerchantWnd")							// not interacting with vendor
+		&& !WindowOpen("TradeWnd")								// not trading with someone
+		&& !WindowOpen("BigBankWnd") && !WindowOpen("BankWnd")	// not banking
+		&& !WindowOpen("LootWnd")								// not looting
+		&& pLocalPC->pSpawn->StandState != STANDSTATE_FEIGN		// not Feigned
+		&& (bIgnoreSafeZones ? !IsSafeZone(iZoneID) : true)		// if we are ignoring safe zones, make sure we're not in one
```

Comment:
This seems a little weird to me here. I'd probably write it as:
```
&& (!bIgnoreSafeZones || !IsSafeZone(iZoneID))
```

Rather than using a ternary and the "true"